### PR TITLE
docs: Update kernel requirements for advanced features

### PR DESCRIPTION
**Pull request by pchaigno**
The first commit rewords a bit the paragraph and title. The second adds three missing advanced features (BPF-based TProxy, bandwidth manager, and session affinity) and updates the IPv4 fragment tracking entry with a link.

### CODE REVIEW

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
I think that only v1.8 is relevant for backporting here, and only for the session affinity changes. Bandwidth manager and tproxy are v1.9 features only.

File: Documentation/install/system_requirements.rst [hunk 1]
```diff
@@ -151,22 +151,25 @@ by adding the following to the helm configuration command line:
...
+
+============================= ===============================
+Cilium Feature                Minimum Kernel Version
+============================= ===============================
+:ref:`concepts_fragmentation` >= 4.10
+:ref:`cidr_limitations`       >= 4.11
+:ref:`host-services`          >= 4.19.57, >= 5.1.16,  >= 5.2
+:ref:`kubeproxy-free`         >= 4.19.57, >= 5.1.16,  >= 5.2
+:ref:`bandwidth-manager`      >= 5.1
+:ref:`session-affinity`       >= 5.7
```
Comment (brb) [inline Documentation/install/system_requirements.rst:170]: We do have a fallback mode for the session-affinity when running on <5.7 kernel:
> The session affinity support is enabled by default for Cilium’s kube-proxy replacement. For users who run on older kernels which do not support the network namespace cookies, a fallback in-cluster mode is implemented, which is based on a fixed cookie value as a trade-off. This makes all applications on the host to select the same service endpoint for a given service with session affinity configured. To disable the feature, set config.sessionAffinity=false.

https://docs.cilium.io/en/v1.8/gettingstarted/kubeproxy-free/#session-affinity

Comment (pchaigno) [reply Documentation/install/system_requirements.rst:170]: So the downside for users of not having network namespace cookies is that all pods on a given node will be serviced by the same backend?

That's a bit hard to convey in a single phrase for the system requirements list. Maybe we could say "Per-namespace session affinity >= 5.7"? Or just fallback to "Full support for session affinity >= 5.7"?

Comment (brb) [reply Documentation/install/system_requirements.rst:170]: > So the downside for users of not having network namespace cookies is that all pods on a given node will be serviced by the same backend?

That's correct, but only for E-W traffic.

> Maybe we could say "Per-namespace session affinity >= 5.7"? Or just fallback to "Full support for session affinity >= 5.7"?

Hmm, `per-namespace session affinity` might confuse users, so maybe just `full support for session affinity` is OK?

Comment (pchaigno) [reply Documentation/install/system_requirements.rst:170]: Please see #13318.

Comment (brb) [reply Documentation/install/system_requirements.rst:170]: Thanks!

**Pull request merged**